js/app/modules/contentGroup/contentGroup.js

```diff
@@ -217,6 +227,30 @@ const ContentGroup = new Module.Class({
         if (!this._selection.in_error_state)
             return;
         this._stack.visible_child_name = ERROR_PAGE_NAME;
+        this._log_button_stack.visible_child_name = 'button';
+    },
+
+    _on_log_button_click: function () {
+        let timestamp = new Date(Date.now());
+        let stamp = 'eos-knowledge-lib log ' + timestamp.toISOString() + '.txt';
+        let log_name = GLib.build_filenamev([GLib.get_home_dir(), stamp]);
```

Comment:
Two comments about this:

a) we probably need to add a proper --filesystem entry to the flatpak manifest, or this won't work.

b) best case scenario the user will never get to see these log files, but, if for whatever reason this occurs (often enough) we could consider using a sub-directory (e.g. ~/debug/) so we don't fill the user's home directory with log files?